cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperator.java

```diff
@@ -257,283 +226,350 @@ protected KafkaConnectStatus createStatus() {
-     * @param dockerFileConfigMap   ConfigMap with the generated Dockerfile
-     * @param newBuildRevision      New build revision (hash of the Dockerfile)
+     * Generates a hash from the trusted TLS certificates that can be used to spot if it has changed.
      *
-     * @return                      Future which completes when the build is finished (or fails if it fails)
+     * @param namespace          Namespace of the Connect cluster
+     * @param kafkaConnectSpec   KafkaConnectSpec object
+     * @return                   Future for tracking the asynchronous result of generating the TLS auth hash
      */
-    private Future<Void> kubernetesBuildStart(Reconciliation reconciliation, String namespace, KafkaConnectBuild connectBuild, ConfigMap dockerFileConfigMap, String newBuildRevision)  {
-        return configMapOperations.reconcile(reconciliation, namespace, KafkaConnectResources.dockerFileConfigMapName(connectBuild.getCluster()), dockerFileConfigMap)
-                .compose(ignore -> serviceAccountOperations.reconcile(reconciliation, namespace, KafkaConnectResources.buildServiceAccountName(connectBuild.getCluster()), connectBuild.generateServiceAccount()))
-                .compose(ignore -> podOperator.reconcile(reconciliation, namespace, KafkaConnectResources.buildPodName(connectBuild.getCluster()), connectBuild.generateBuilderPod(pfa.isOpenshift(), imagePullPolicy, imagePullSecrets, newBuildRevision)))
-                .mapEmpty();
+    Future<Integer> generateAuthHash(String namespace, KafkaConnectSpec kafkaConnectSpec) {
+        KafkaClientAuthentication auth = kafkaConnectSpec.getAuthentication();
+        List<CertSecretSource> trustedCertificates = kafkaConnectSpec.getTls() == null ? Collections.emptyList() : kafkaConnectSpec.getTls().getTrustedCertificates();
+        return Util.authTlsHash(secretOperations, namespace, auth, trustedCertificates);
     }
 
     /**
-     * Checks if the builder Pod finished the build
+     * Creates the annotations for the KafkaConnect deployment
      *
-     * @param namespace             Namespace where the Kafka Connect is deployed
-     * @param podName               The name of the Pod which should be checked whether it finished
-     *
-     * @return                      True if the build already finished, false if it is still building
+     * @param buildState           State of the Connect build
+     * @param desiredLogging       Atomic reference to the desired logging string
+     * @param hash                 Tls certificate hash
+     * @return                     Map of keys and values for Connect deployment annotations
      */
-    private boolean kubernetesBuildPodFinished(String namespace, String podName)   {
-        Pod buildPod = podOperator.get(namespace, podName);
-        return KafkaConnectBuildUtils.buildPodComplete(buildPod);
+    Map<String, String> createAnnotations(BuildState buildState, AtomicReference<String> desiredLogging, Integer hash) {
```

Comment:
I was looking at it from the viewpoint that it's hard to discern what annotations are currently added to the deployment. If I were to add more annotations I would want to see what else is being added to decide what makes sense for me to add. I was also trying to reduce the number of objects that have to be held outside the compose loop so that as you scan down the code you have less variables to keep track of. But I do see your point about the annotations being related to the code. I'll take another look.